services/director-v2/src/simcore_service_director_v2/utils/rabbitmq.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any
+
+from models_library.projects import ProjectID
+from models_library.projects_nodes_io import NodeID
+from models_library.projects_state import RunningState
+from models_library.rabbitmq_messages import (
+    InstrumentationRabbitMessage,
+    RabbitResourceTrackingHeartbeatMessage,
+    RabbitResourceTrackingStartedMessage,
+    RabbitResourceTrackingStoppedMessage,
+    SimcorePlatformStatus,
+)
+from models_library.services import ServiceKey, ServiceType, ServiceVersion
+from models_library.services_resources import ServiceResourcesDict
+from models_library.users import UserID
+from models_library.wallets import WalletID
+from servicelib.rabbitmq import RabbitMQClient
+
+from ..models.comp_tasks import CompTaskAtDB
+
+
+async def publish_service_started_metrics(
+    rabbitmq_client: RabbitMQClient,
+    *,
+    user_id: UserID,
+    simcore_user_agent: str,
+    task: CompTaskAtDB,
+) -> None:
+    message = InstrumentationRabbitMessage.construct(
+        metrics="service_started",
+        user_id=user_id,
+        project_id=task.project_id,
+        node_id=task.node_id,
+        service_uuid=task.node_id,
+        service_type=task.node_class.value,
+        service_key=task.image.name,
+        service_tag=task.image.tag,
+        simcore_user_agent=simcore_user_agent,
+    )
+    await rabbitmq_client.publish(message.channel_name, message)
+
+
+async def publish_service_stopped_metrics(
+    rabbitmq_client: RabbitMQClient,
+    *,
+    user_id: UserID,
+    simcore_user_agent: str,
+    task: CompTaskAtDB,
+    task_final_state: RunningState,
+) -> None:
+    message = InstrumentationRabbitMessage.construct(
+        metrics="service_stopped",
+        user_id=user_id,
+        project_id=task.project_id,
+        node_id=task.node_id,
+        service_uuid=task.node_id,
+        service_type=task.node_class.value,
+        service_key=task.image.name,
+        service_tag=task.image.tag,
+        result=task_final_state,
+        simcore_user_agent=simcore_user_agent,
+    )
+    await rabbitmq_client.publish(message.channel_name, message)
+
+
+async def publish_service_resource_tracking_started(  # noqa: PLR0913
```

Comment:
it would be great if these were part of some library so that I could reuse them in the dy-sidecar.